src/controls/MenuSelectHeading.tsx

```diff
@@ -200,6 +194,15 @@ export default function MenuSelectHeading({
     return new Set(headingExtension?.options.levels ?? []);
   }, [editor]);
 
+  // We have to pass a level when running `can`, so this is just an arbitrary
+  // one. And we have to check `currentLevel` to prevent all other heading
+  // levels from being disabled when Heading 1 is selected (see
+  // https://github.com/sjdemartini/mui-tiptap/issues/197).
+  const canSetHeading =
+    enabledHeadingLevels.size > 0 &&
+    (currentLevel === [...enabledHeadingLevels][0] ||
```

Comment:
To make this more performant, can you pull out `[...enabledHeadingLevels][0]` into a local variable instead of repeating it twice? And for getting the value, can you update this to use `enabledHeadingLevels.values().next()`? The list spread approach is more expensive (e.g. see [here](https://stackoverflow.com/questions/32539354/how-to-get-the-first-element-of-set-in-es6-ecmascript-2015#comment122997793_32539929) for an explanation)